ts/src/kucoin.ts

```diff
@@ -1001,65 +1002,37 @@ export default class kucoin extends Exchange {
-        //        "code": "200",
-        //        "msg": "success",
-        //        "retry": false,
-        //        "data": [
-        //            {
-        //                "withdrawMinFee": "0.0005",
-        //                "chainName": "BTC",
-        //                "preDepositTipEnabled": "false",
-        //                "chain": "btc",
-        //                "isChainEnabled": "true",
-        //                "withdrawDisabledTip": "",
-        //                "walletPrecision": "8",
-        //                "chainFullName": "Bitcoin",
-        //                "orgAddress": "",
-        //                "isDepositEnabled": "true",
-        //                "withdrawMinSize": "0.001",
-        //                "depositDisabledTip": "",
-        //                "userAddressName": "",
-        //                "txUrl": "https://blockchain.info/tx/{txId}",
-        //                "preWithdrawTipEnabled": "false",
-        //                "withdrawFeeRate": "0",
-        //                "confirmationCount": "2",
-        //                "currency": "BTC",
-        //                "depositMinSize": "0.00005",
-        //                "isWithdrawEnabled": "true",
-        //                "preDepositTip": "",
-        //                "preWithdrawTip": "",
-        //                "status": "enabled"
-        //            },
-        //        ]
+        //        "code":"200000",
+        //        "data":[
+        //           {
+        //              "currency":"CSP",
+        //              "name":"CSP",
+        //              "fullName":"Caspian",
+        //              "precision":8,
+        //              "confirms":null,
+        //              "contractAddress":null,
+        //              "isMarginEnabled":false,
+        //              "isDebitEnabled":false,
+        //              "chains":[
+        //                 {
+        //                    "chainName":"ERC20",
+        //                    "chain":"eth",
+        //                    "withdrawalMinSize":"2999",
+        //                    "withdrawalMinFee":"2999",
+        //                    "isWithdrawEnabled":false,
+        //                    "isDepositEnabled":false,
+        //                    "confirms":12,
+        //                    "preConfirms":12,
+        //                    "contractAddress":"0xa6446d655a0c34bc4f05042ee88170d056cbaf45"
+        //                 }
+        //              ]
+        //           },
         //    }
         //
-        const responses = await Promise.all (promises);
-        const responseCurrencies = responses[0];
-        const responseChains = responses[1];
-        const data = this.safeValue (responseCurrencies, 'data', []);
-        const chainsData = this.safeValue (responseChains, 'data', []);
-        const currencyChains = this.groupBy (chainsData, 'currency');
+        const response = await this.publicGetCurrencies (params);
```

Comment:
@ttodua can you check if you think that we can replace the internal API with this newly released v3 endpoint?

Reply:
there seems to be some imperfections in new api, bcz in internal data: https://www.kucoin.com/_api/currency/currency/chain-info
there was "walletPrecision" for each network, for example for USDT on ALGORAND network, the precision was 6, but for EOS network it was  4. in new api, each currency has only one precision (even though it might have 10 networks). so that's missing part.  in addition, there was some informative fields in internal api (`'preDepositTip', 'depositDisabledTip',...` useful info for users) but ofc we should use official endpoint, and can leave that "internal api" call in promise-all, with only 1 retry. will not hurt imo.

Reply:
added https://github.com/carlosmiei/ccxt/pull/12/files

Reply:
@ttodua I think it is reasonable yes, nonetheless will ask them if they can add those fields to the public API it would be easier

Reply:
> @ttodua I think it is reasonable yes, nonetheless will ask them if they can add those fields to the public API it would be easier

I've posted in their api group: https://t.me/KuCoin_API/172723 and those fields are needed. preferably, if they want to avoid changing the api response, they can add a request-param to exlicitly add those fields in response, so no-one existing user will be affected.